### DISCUSSION TITLE
truncation for diagonal UniTensor

### DISCUSSION THREAD

**Discussion by hunghaoti** [Cytnx-dev/Cytnx#304] [Q&A]
When create a diagonal UniTensor with shape [3,3] and then call truncate(idx=0, dim=2) to truncate the bond dimension for first bond. Currently, it will return a diagonal UniTensor with shape [2,2]. Is it expected or a bug ?

The related unit test is and see line #L4562
https://github.com/Cytnx-dev/Cytnx/blob/f62f0cedd6197f7c68999dd2f814783c4b40cc8e/tests/DenseUniTensor_test.cpp#L4545-L4566

**Comment by yingjerkao** [comment]
I think it is expected but misleading. Maybe we should throw a warning so that the user may not misuse it.

**Comment by ianmccul** [comment]
Diagonal matrices don't have to be square; there is no conceptual problem to having a $m \times n$ matrix with only diagonal elements and $m \neq n$.

**Reply by manuschneider** [reply]
I agree, it would be better to allow for non-square diagonal matrices
